File: src/Link/Link.ReportManagement.Domain.Model/Entities/Event.cs
using System;
using Link.Common.Domain.Framework.Models;
using Link.ReportManagement.Domain.Model.Enums;
using System.Collections.Generic;

namespace Link.ReportManagement.Domain.Model.Entities
{
    public class Event : Entity<Event>
    {
        public Event(
            string id,
            UserId userId,
            string name,
            ExpertType expertType,
            ExpertStatus status,
            int countOfNeededExperts,
            List<ExpertId> experts, 
            double latitude,
            double longitude,
            DateTime startTime, 
            DateTime endTime)
        {
            Id = id;
            UserId = userId;
            Name = name;
            ExpertType = expertType;
            Status = status;
            Latitude = latitude;
            Longitude = longitude;
            StartTime = startTime;
            EndTime = endTime;
            CountOfNeededExperts = countOfNeededExperts;
            Experts = new HashSet<ExpertId>(experts);
        }

        public string Id { get; }

        public UserId UserId { get; }

        public string Name { get; }

        public ExpertType ExpertType { get; }

        public ExpertStatus Status { get; }

        public DateTime StartTime { get; }

        public DateTime EndTime { get; }

        public double Latitude { get; }

        public double Longitude { get; }

        public int CountOfNeededExperts { get; }

        public HashSet<ExpertId> Experts { get; }
    }
}

File: src/Link/Link.Frontend/Link-Frontend/src/app/models/event.ts
import { Expert } from './expert';
import { EventDto } from '../interfaces/event-dto';

export class Event {
    constructor(
        public id: string,
        public userId: string,
        public name: string,
        public expertType: string,
        public status: string,
        public latitude: number,
        public longitude: number,
        public startTime: Date,
        public endTime: Date,
        public countOfNeededExperts: number,
        public experts: Expert[]
    ) { }

    public static Create(dto: EventDto): Event {
        // tslint:disable-next-line: prefer-const
        let parsedExperts: Expert[];

        if (dto.experts != null && dto.experts.length !== 0) {
            dto.experts.forEach(expert => {
                parsedExperts.push(Expert.Create(expert));
            });
        }

        return new Event(
            dto.id,
            dto.userId,
            dto.name,
            dto.expertType,
            dto.status,
            dto.latitude,
            dto.longitude,
            dto.startTime,
            dto.endTime,
            dto.countOfNeededExperts,
            parsedExperts);
    }
}
